src/utils.ts

```diff
@@ -1,10 +1,21 @@
 import { readFileSync, writeFileSync } from 'fs';
 import { Buffer } from 'buffer';
 import Storable from "./Storable";
-import { Key } from './Config';
+import { ConfigOptions, DEFAULT_CONFIG, Key } from './Config';
 
-export function sync(file: string, data: Record<string, unknown>): void {
-  writeFileSync(file, JSON.stringify(data));
+export function sync(
+  file: string,
+  data: Record<string, unknown>,
+  options: ConfigOptions = DEFAULT_CONFIG,
```

Comment:
This function could work with only the sub object `prettyJson` instead of the whole thing.
That would require to split 
```
export type ConfigOptions = {
     prettyJson?: {
         enabled: boolean;
         indentSize?: number;
     };
 };
```
into 
```
export type PrettyJSONConfig = export type ConfigOptions = {
    enabled: boolean;
    indentSize?: number;
};
export type ConfigOptions = {
     prettyJson?: PrettyJSONConfig;
};
```

Reply:
this would also simplify
```
if (options.prettyJson && options.prettyJson.enabled)
```
into 
```
if (options?.enabled)
```